src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1263,12 +1262,29 @@ protected override void ApplyDocumentInfoChanged(DocumentId documentId, Document
             // By setting this property, Visual Studio will perform the file rename, which 
             // will cause the workspace's current solution to update and will fire the 
             // necessary workspace changed events.
-            projectItemForDocument.Name = uniqueName;
+            try
+            {
+                projectItemForDocument.Name = uniqueName;
+            }
+            catch (InvalidOperationException ex)
+            {
+                FatalError.ReportAndCatch(ex);
+                return;
+            }
 
-            if (projectItemForDocument.TryGetFullPath(out var newPath))
+            string? newPath;
+            try
             {
-                undoManager?.Add(new RenameDocumentUndoUnit(this, uniqueName, document.Name, newPath));
+                projectItemForDocument.TryGetFullPath(out newPath);
             }
+            catch (KeyNotFoundException ex)
```

Comment:
i wasn't sure if i should move this into TryGetFullPatah (as that's an extension method we own).  this feels like something messed up in the project system that we need this.

Note: i can provide you with file paths this happens for if that would be helpful.